### ISSUE TITLE
Edit config error (number of specified <arg> doesn't match the number of GPUs)

### ISSUE THREAD

**Issue by sebas05000** [trexminer/T-Rex#1170]
Hi,

Not sure why but I am getting this error after I want to enable 2 more GPUs on the WebUI.
![image](https://user-images.githubusercontent.com/66391171/158878891-6edd894e-8bd3-48c1-bab1-2deef6d2f1ad.png)

The GPUs I have are all NVIDIA 4 are 3090 and the 2 I want to enable are 3080ti

**Comment by pdieppa** [comment]
From the manual

-d, --devices                  Comma separated list of CUDA devices to use.
                                   Device IDs start counting from 0.

**Comment by sebas05000** [comment]
> From the manual
> 
> -d, --devices Comma separated list of CUDA devices to use. Device IDs start counting from 0.

The device IDs are fine since I use a config file and edit everything on the WebUI never had this issue before

**Comment by pdieppa** [comment]
I don't use the WebUI, but usually that error could be the number of devices specified in the program do not match the actual number of cards detected by the system. Also to keep in mind that by default, the IDs start with 0 and not with 1, but I'm sure you already know that. :)

**Comment by sebas05000** [comment]
> I don't use the WebUI, but usually that error could be the number of devices specified in the program do not match the actual number of cards detected by the system. Also to keep in mind that by default, the IDs start with 0 and not with 1, but I'm sure you already know that. :)

Yes ik about the IDs starting from 0 and not with 1 and that is weird on the WebUI it shows all devices fine that i have present on the PC only that I cannot enable more than 4 even so their IDs look fine as you can see here:
![image](https://user-images.githubusercontent.com/66391171/159140372-1ef061a5-2866-4a09-8e80-56b5e410fc08.png)
![image](https://user-images.githubusercontent.com/66391171/159140376-3ad66c0e-a705-469d-93f4-a9d0511cc15b.png)

**Comment by sebas05000** [comment]
Any help here?

**Comment by pdieppa** [comment]
I don't know what those screens are from but I would look at the configuration file to make sure the device switch is set correctly. From the error it looks like there are too many IDs for the number of GPUs you have.  Does this make sense?
